### PR TITLE
[pigeon] Null safe requires Dart 2.12

### PR DESCRIPTION
**Pull request by blasten**
This is required after the beta announce.

### CODE REVIEW

**Review by gaaclarke — APPROVED** [review]
Looks good, thanks!  pPease update the `pubspec.yaml` and `generator_tools.dart`.

**Comment by gaaclarke** [comment]
Any idea what's going on with that test?  Looks like there is a nil value for `cached_framework_dir`.

https://github.com/flutter/packages/blob/0e3e175bebed42c36a280bcd90cc26b8c0376931/packages/pigeon/e2e_tests/test_objc/ios/Podfile#L55

I didn't write that Podfile, it was either autogenerated by flutter or provided by e2e somehow.

File: .cirrus.yml [hunk 1]
```diff
@@ -70,14 +70,20 @@ task:
   use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true'
   name: local_tests
   osx_instance:
-    image: catalina-flutter
+    image: high-sierra-xcode-9.4.1
```
Comment (xster) [inline .cirrus.yml:73]: are changes to this file intentional?

Comment (blasten) [reply .cirrus.yml:73]: No. This was for debugging purposes.

**Comment by blasten** [comment]
@gaaclarke I am able to repro this issue locally: https://github.com/flutter/packages/runs/1809227404  - Do you know what might be causing it?

**Comment by gaaclarke** [comment]
Yep, you have to remove xcpretty from the local_tests.  I did it in an external contributors PR yet but it hasn't landed yet.

**Comment by blasten** [comment]
🎉 finally

**Pull request merged**